semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -83,7 +83,14 @@ exception InvalidLanguage of Rule.rule_id * string * Parse_info.t
  * start of the pattern *)
 exception
   InvalidPattern of
-    Rule.rule_id * string * Rule.xlang * string (* exn *) * Parse_info.t
+    Rule.rule_id
```

Comment:
A record type could be nice here too.

Reply:
I'll leave this for now because the exception isn't used in that many places and I'd want to change all of them for consistency